### PR TITLE
Nuke User owned AMIs

### PR DESCRIPTION
**Pull request by tonerdo**
This PR does the following:

- Adds support for listing and nuking Amazon Machine Images owned by the user
- Adds tests for this functionality

### CODE REVIEW

File: aws/ami.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+	logging.Logger.Infof("Terminating all AMIs in region %s", *session.Config.Region)
+
+	for _, imageID := range imageIds {
+		params := &ec2.DeregisterImageInput{
+			ImageId: imageID,
+		}
+
+		_, err := svc.DeregisterImage(params)
+		if err != nil {
```
Comment (brikis98) [inline aws/ami.go]: If you try to delete an AMI that's actively in use, do you get an error? If so, we should catch it, log it, and keep going.

File: aws/ami_test.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/aws-nuke/util"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func createTestAMI(t *testing.T, session *session.Session, name string) ec2.Image {
+	svc := ec2.New(session)
+	instance := createTestEC2Instance(t, session, name, false)
```
Comment (brikis98) [inline aws/ami_test.go]: Should this `defer` the deletion of this instance?

Comment (tonerdo) [reply aws/ami_test.go]: It does the defer on line 62. If the `defer` is done here, the EC2 instance could have been nuked before we get a chance to create the Image. The safest bet is to just clean up after the test function itself

File: aws/ami.go (hunk 1 above)
Comment (tonerdo) [reply aws/ami.go]: Will try this out

Comment (tonerdo) [reply aws/ami.go]: From my tests an AMI can be deleted even when there exists an EC2 instance that was created with it. I believe once the VM is provisioned with that machine image there's no link between them and that ami is exists just for re-provisioning new EC2 instances

Comment (brikis98) [reply aws/ami.go]: Good to know!

Ship it 👍

Comment (tonerdo) [reply aws/ami.go]: Sweet, I'll merge once I fix the test

Comment (josh-padnick) [reply aws/ami.go]: @tonerdo Is deleting an AMI actually a two-step procedure as [I had to do](https://github.com/josh-padnick/ec2-snapper/blob/master/delete_command.go#L258) in ec2-snapper? At that time, I first had to deregister the AMI (as you do here), and second, had to delete the AMI's corresponding EBS snapshot.

**Pull request merged**